tools/sesman/sesmanworker/targetaccuratefilter.py

```diff
@@ -0,0 +1,107 @@
+"""
+
+Filter field by field by applying one or several filters
+on target string like account@domain@device:service
+
+For active the target accurate filter mode, user MUST type on
+selector research a string form beginning by '$' character like : 
+"$keyword1:value1,$keyword2:value2,$keyword3:value3"
+
+Each wanted filter must following the "$keyword:value" 
+format for be applicable.
+
+For example, the following target filter string will filter on
+account, domain, device and service fields 
+according to the passed values : 
+"$account:my_account,$domain:my_domain,$device:my_device,$service:my_service"
+    
+We can combine 2 filters or more like above but we 
+can also execute only one filter on target.
+The following target filter strings are accepted :
+"$account:my_account"
+"$domain:my_domain"
+"$device:my_device"
+"$service:my_service"
+
+There is no specifc order for apply filters. 
+The following target filter string is also valid :
+"$device:my_device,$account:my_account,$domain:my_domain"
+
+However, target filter string badly formatted like containing 
+duplicated filter keywords, missing values and others aren't 
+allowed and will be skipped and considered as no valid.
+The following examples are forbidden :
+"$account:my_account,$account:my_account"
+"$domain:my_domain,"
+"$my_device:my_device"
+"$service:"
+
+"""
+
+
+class FilterKeywordSyntaxError(RuntimeError) :
+    def __init__(self, keyword) :
+        super().__init__(
+            self, ("unknown filter keyword with '%s'"
+                   % keyword))
+
+class FilterKeywordDuplicateError(RuntimeError) :
+    def __init__(self, keyword) :
+        super().__init__(
+            self, ("duplicated filter keyword with '%s'"
+                   % keyword))
+
+class ParsingError(RuntimeError) :
+    def __init__(self, bad_string) :
+        super().__init__(
+            self, ("'keyword:value' format parsing error with '%s'"
+                   % bad_string))
+
+
+
+def is_filter_keyword(filter_keyword) :
+    if filter_keyword.startswith('$') :
+        filter_keyword_no_prefix = filter_keyword[1:]
+        
+        if (filter_keyword_no_prefix in ("account",
+                                         "domain",
+                                         "device",
+                                         "service")) :
+            return True, filter_keyword_no_prefix
+        return False, None
```

Comment:
remove one level of indentation on the return line so that the function always returns tuple.
Do you really needs tuple ? string value that can be None might be enough.

Reply:
Tuple was for check if `filter_keyword_no_prefix` is gettable and not `None` but you're right, just return `filter_keyword_no_prefix` or `None` is enough